arangod/Aql/AqlValue.cpp

```diff
@@ -1093,10 +1276,47 @@ int AqlValue::Compare(velocypack::Options const* options, AqlValue const& left,
     }
     // fall-through to other types intentional
   }
-
   // if we get here, types are equal or can be treated as being equal
 
   switch (leftType) {
+    case VPACK_INLINE_INT48:
+    case VPACK_INLINE_UINT48:
+    case VPACK_INLINE_INT64:
+    case VPACK_INLINE_UINT64:
+    // if right value type is aslo optimized inline we can optimize comparison
+      if (leftType == rightType) {
+        if (leftType == VPACK_INLINE_UINT64) {
+          uint64_t l = static_cast<uint64_t>(left.toInt64());
+          uint64_t r = static_cast<uint64_t>(right.toInt64());
+          if (l == r) {
+            return 0;
+          }
+          return (l < r ? -1 : 1);
+        } else {
+          int64_t l = left.toInt64();
+          int64_t r = right.toInt64();
```

Comment:
What happens here if `right` is a `uint64_t`?

Reply:
It can`t happen. If we got here - then at line 1287 we checked that leftType is equal to rightType. So here we know that teftType and rightType are both equal and both are not UINT64